test/unit/queryNotificationHandler.test.ts

```diff
@@ -28,165 +27,114 @@ suite("QueryNotificationHandler tests", () => {
     let queryCompleteHandler: NotificationHandler<any>;
 
     setup(() => {
+        sandbox = sinon.createSandbox();
         notificationHandler = new QueryNotificationHandler();
         eventData = { ownerUri: "testUri" };
 
-        // Setup mock - Use the same QueryRunner for the whole test - this tests if it can be reused
-        runnerMock = TypeMoq.Mock.ofType(QueryRunner, TypeMoq.MockBehavior.Loose);
-        runnerMock.callBase = true;
-        runnerMock
-            .setup((x) => x.handleBatchStart(TypeMoq.It.isAny()))
-            .callback((event) => {
-                batchStartHandlerCalled = true;
-            });
-        runnerMock
-            .setup((x) => x.handleMessage(TypeMoq.It.isAny()))
-            .callback((event) => {
-                messageHandlerCalled = true;
-            });
-        runnerMock
-            .setup((x) => x.handleResultSetComplete(TypeMoq.It.isAny()))
-            .callback((event) => {
-                resultSetCompleteHandlerCalled = true;
-            });
-        runnerMock
-            .setup((x) => x.handleBatchComplete(TypeMoq.It.isAny()))
-            .callback((event) => {
-                batchCompleteHandlerCalled = true;
-            });
-        runnerMock
-            .setup((x) => x.handleQueryComplete(TypeMoq.It.isAny()))
-            .callback((event) => {
-                queryCompleteHandlerCalled = true;
-                runnerMock.object.setHasCompleted();
-            });
-
-        // Get handlers
+        runnerMock = sandbox.createStubInstance(QueryRunner);
+        runnerMock.handleQueryComplete.callsFake(() => {
+            runnerMock.setHasCompleted();
+        });
+
+        runner = runnerMock as unknown as QueryRunner;
```

Comment:
Same question as before about casting to unknown as an intermediary cast. Just commenting in case this needs to change.